pkg/reconciler/taskrun/taskrun.go

```diff
@@ -360,11 +360,15 @@ func (c *Reconciler) reconcile(ctx context.Context, tr *v1beta1.TaskRun,
 			tr.Spec.Workspaces = taskRunWorkspaces
 		}
 
+		willOverwriteAffinity := podconvert.WillOverwritePodSetAffinity(tr)
 		pod, err = c.createPod(ctx, tr, rtr)
```

Comment:
is it possible to move in the `willOverwritePodSetAffinity` to inside `c.createPod()` - not sure it is possible, but it would be closer to the place where the affinity is set.

Reply:
Thank you for the review. Yes, it can be moved inside `createPod`, but then we need to pass the recorder to `createPod` as well, and it will need to start emitting events. I implemented it this way to avoid changing `createPod` signature, and to keep all the event emitting in the `reconciler` package. 

If moving this to the `pod` package is preferred, please let me know and I'll update the PR.

Reply:
It looks like you wouldn't have to change the signature of `createPod`, you can look up the recorder from the context:

```go
// in c.createPod
recorder := controller.GetEventRecorder(ctx)
if recorder != nil && willOverwritePodAffinity(tr) {
    recorder.Eventf(...)
}
```

Reply:
Thank you for the reviews. Moved `willOverwritePodSetAffinity` to `createPod`, no changes were made to the signature of `createPod`.